spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/db/migration/AbstractCallback.java

```diff
@@ -67,7 +68,18 @@ public boolean canHandleInTransaction(Event event, Context context) {
 
 	@Override
 	public void handle(Event event, Context context) {
-		runner.execute(context.getConnection(), getCommands(event, context));
+		try {
+			runner.execute(context.getConnection(), getCommands(event, context));
+		}
+		catch(Exception sqe) {
+			if (sqe instanceof BadSqlGrammarException && sqe.getMessage().contains("already exists")) {
```

Comment:
I'm not sure if `already exists` is returned with every database as only reliable way is to check sql error code.